### DISCUSSION TITLE
Renderer process without contextBridge

### DISCUSSION THREAD

**Discussion by jsheffers** [cawa-93/vite-electron-builder#470] [Q&A]
Is it possible to use this template without `contextBridge` and with `contextIsolation: false`?  I know this is not recommended, however, the software I build is for internal purposes only, and the machines they are being installed on are highly regulated (only inside the building and locked down from a domain controller).

**Comment by cawa-93** [comment]
Context isolation means that `window` in `renderer` is not same `window` in `preload`. I think this is not what you want. Maybe you should check [`@electron/remote`](https://www.npmjs.com/package/@electron/remote)

**Reply by jsheffers** [reply]
I cannot enable `contextIsolation` because of the following:  https://github.com/electron/electron/issues/16513#issuecomment-816117930. In my current implementation using `electron-webpack` this is not an issue.  I'm able to have contextIsolation false and not use the contextBridge.  Does Vite have a limitation here?

**Reply by cawa-93** [reply]
You can disable context isolation. And set global methods like that
https://www.electronjs.org/docs/tutorial/context-isolation#before-context-isolation-disabled

**Reply by jsheffers** [reply]
That's exactly what I'm doing and it works in my `electron-webpack` environment. As soon as I try to import ipcRenderer in my renderer process using this template, I get the path.join is not a function error.

**Reply by cawa-93** [reply]
May you show code?

**Reply by jsheffers** [reply]
If I take your default template and make two changes I can replicate this.  Alter `main/src/index.ts` to turn `nodeIntegration: true` and `contextIsolation: false` for the mainWindow:

```javascript
  mainWindow = new BrowserWindow({
    show: false, // Use 'ready-to-show' event to show window
    webPreferences: {
      contextIsolation: false, // Added by me
      nodeIntegration: true, // Added by me
      nativeWindowOpen: true,
      preload: join(__dirname, '../../preload/dist/index.cjs'),
    },
  });
```
Then replace `About.vue` with:

```javascript
<template>
  <h2 id="versions">
    Lib versions
  </h2>
</template>

<script lang="ts">
import {defineComponent} from 'vue';
import {ipcRenderer} from 'electron';
export default defineComponent({
  name: 'App',
  setup() {
    ipcRenderer.on('something', () => {
      console.log('something');
    });
  },
});
</script>
```

Run `yarn watch`, click the about link, then check the devtools console.

**Reply by cawa-93** [reply]
The reason you can directly import other packages is because Vite bundle all dependencies into your code. That's why actually you don't have any imports in runtime. And technically (if nodeIntegration is disabled) and you can't import anything.

However, the electron package is exception. It cannot be bundled.

So Yes, to take access to electron package in `renderer` module you need enable nodeIntegration. You can access to electron package in preload module without nodeIntegration.

**Reply by jsheffers** [reply]
> So Yes, to take access to electron package in renderer module you need to enable nodeIntegration.

My point was that I have nodeIntegration set to true and I still can't access it in the renderer (see my example). I can however access it through the preload.  Am I missing something?

**Reply by ssj17vegeta** [reply]
Don't know if that can help you @jsheffers , but one thing I overlooked at the beginning though it was mentioned in the docs is you cannot use the "import" syntax, you have to use the "require" syntax to be able to use path, electron, fs packages in the renderer.

So instead of
`import { something } from 'electron'`
it would be
`const something = require('electron').something`
